lxd/container_test.go

```diff
@@ -190,6 +190,33 @@ func (suite *containerTestSuite) TestContainer_LogPath() {
 	suite.Req.Equal(shared.VarPath("logs", "testFoo"), c.LogPath())
 }
 
+func (suite *containerTestSuite) TestContainer_Devlxd_Exists() {
+	args := db.ContainerArgs{
+		Ctype:     db.CTypeRegular,
+		Ephemeral: false,
+		Config:    map[string]string{"security.devlxd": "true"},
+		Name:      "testFoo",
+	}
+
+	c, err := containerCreateInternal(suite.d.State(), args)
+	suite.Req.Nil(err)
+	suite.Req.True(c.IsDevLxd_Enabled(), "This container should have /dev/lxd enabled.")
+	suite.Req.Nil(c.Delete(), "Failed to delete the container.")
+}
+
+func (suite *containerTestSuite) TestContainer_Devlxd_Not_Exists() {
+	args := db.ContainerArgs{
+		Ctype:     db.CTypeRegular,
+		Ephemeral: false,
+		Config:    map[string]string{"security.devlxd": "false"},
+		Name:      "testFoo",
+	}
+	c, err := containerCreateInternal(suite.d.State(), args)
+	suite.Req.Nil(err)
+	suite.Req.False(c.IsDevLxd_Enabled(), "This container should have /dev/lxd disabled.")
+	suite.Req.Nil(c.Delete(), "Failed to delete the container.")
+}
```

Comment:
This test isn't very useful as all it does is check that when a string is set in the database that the same string is returned back from it. We have generic DB tests for that, so this doesn't really cover anything useful.

Instead, what would be useful is an integration test. Something like this (untested):
```diff
diff --git a/test/suites/devlxd.sh b/test/suites/devlxd.sh
index ea1a6e88..00b237b3 100644
--- a/test/suites/devlxd.sh
+++ b/test/suites/devlxd.sh
@@ -7,7 +7,12 @@ test_devlxd() {
   # shellcheck disable=SC2164
   cd -
 
-  lxc launch testimage devlxd
+  lxc launch testimage devlxd -c security.devlxd=false
+
+  ! lxc exec devlxd -- test -S /dev/lxd/sock
+  lxc config unset devlxd security.devlxd
+  lxc restart devlxd --force
+  lxc exec devlxd -- test -S /dev/lxd/sock
 
   lxc file push "${TEST_DIR}/devlxd-client" devlxd/bin/
 
```